src/io/itk-dicom/dicom.cpp

```diff
@@ -112,6 +125,89 @@ void movefile(const std::string &src, const std::string &dst) {
   }
 }
 
+// doesn't actually do any length checks, or overflow checks, or anything
+// really.
+template <int N>
+double dotProduct(const std::vector<double> &vec1,
+                  const std::vector<double> &vec2) {
+  double result = 0;
+  for (int i = 0; i < N; i++) {
+    result += vec1.at(i) * vec2.at(i);
+  }
+  return result;
+}
+
+std::string encodeCosinesAsIDPart(const std::vector<double> &cosines) {
```

Comment:
This method is important too since it will help define the key/id for that sub-series. Two notes:
1. It could use a comment explaining that, and summarizing the format/template of the key it produces
2. Could that be a lambda within `RestrictOnImageOrientation` since it's not used anywhere else? Or do you anticipate it will be used outside of that scope?

Reply:
I like the lambda idea, and I'll add a comment on the format.